### PR TITLE
Fix: cssVars API not working correctly on Booker Timeslot selection page

### LINKED ISSUE #8321: [CAL-1505] [CAL-1504] CSS Vars changes don't work on timeslot selection page of Booker
Screenshot 2023-04-17 at 12 30 29 PM

<sub>[CAL-1504](https://linear.app/calcom/issue/CAL-1504/css-vars-changes-dont-work-on-timeslot-selection-page-of-booker)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?
- The reason seems to be that useEmbedUIConfig is being used in _app.tsx and currently Embed flow isn't correctly able to trigger renders for that for SSG pages.
- Because setting CSS Vars doesn't actually need React. It would be more performant to simply use Javascript to create a style tag with those CSS Vars. This would allow us to not spend time fixing that React rendering issue(mentioned above). We can't do SSR for the style as it works through Embed API that works on client.

Fixes #8321

**Environment**: Production

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] Open rick/get-rick-rolled directly in embed(Browsing through profile to the page would work already). You will see that css vars changes aren't applied. It is fixed by this PR.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/public/embed-init-iframe.js [hunk 1]
```diff
@@ -24,3 +24,23 @@ window.getEmbedNamespace = () => {
   const namespace = url.searchParams.get("embed");
   return namespace;
 };
+window.CalEmbed = window.CalEmbed || {};
+window.CalEmbed.applyCssVars = (cssVarsPerTheme) => {
```
Comment (hariombalhara) [inline apps/web/public/embed-init-iframe.js:28]: Verify that Next.js doesn't revert this style being added.

File: apps/web/pages/_app.tsx [hunk 2]
```diff
@@ -28,17 +27,6 @@ const calFont = localFont({
 function MyApp(props: AppProps) {
   const { Component, pageProps, err, router } = props;
   let pageStatus = "200";
-  const { cssVarsPerTheme } = useEmbedUiConfig();
```
Comment (hariombalhara) [inline apps/web/pages/_app.tsx:31]: Handled by JS out of React context

File: apps/web/public/embed-init-iframe.js
Comment (hariombalhara) [inline apps/web/public/embed-init-iframe.js:1]: Moved as a ts file in embed-core.

**Review by hariombalhara — COMMENTED** [review]
Self review done

File: packages/embeds/embed-react/inline.tsx [hunk 3]
```diff
@@ -20,6 +20,19 @@ function App() {
...
       });
+
+      api("ui", {
+        cssVarsPerTheme: {
+          light: {
+            "cal-border-booker": "red",
+            "cal-border-booker-width": "20px",
+          },
+          dark: {
+            "cal-border-booker": "red",
```
Comment (sean-brydon) [inline packages/embeds/embed-react/inline.tsx:31]: Do we default these to red ? Or was this just for testing?

Comment (hariombalhara) [reply packages/embeds/embed-react/inline.tsx:31]: This is actually a test page for inline embed. 
Not clear from the name though. Will rename it in a followup

Comment (sean-brydon) [reply packages/embeds/embed-react/inline.tsx:31]: Gotcha :) Other than that LGTM

**Review by sean-brydon — APPROVED** [review]
LGTM - nice improvement :)

**Pull request merged**